tests/test_manual_connect.c

```diff
@@ -1,19 +1,57 @@
 /*
- * Copyright (c) 2018-2024 Snowflake Computing, Inc. All rights reserved.
- */
+* Copyright (c) 2018-2024 Snowflake Computing, Inc. All rights reserved.
+*/
 
 #include "utils/test_setup.h"
 
-void test_connect_with_duo_push(void **unused)
+/**
+ * Test connection with OAuth authentication.
+ */
+void test_oauth_connect(void **unused) 
 {
     SF_CONNECT *sf = snowflake_init();
     snowflake_set_attribute(sf, SF_CON_ACCOUNT,
                             getenv("SNOWFLAKE_TEST_ACCOUNT"));
     snowflake_set_attribute(sf, SF_CON_USER, getenv("SNOWFLAKE_TEST_USER"));
-    snowflake_set_attribute(sf, SF_CON_PASSWORD,
-                            getenv("SNOWFLAKE_TEST_PASSWORD"));
+
     char *host, *port, *protocol;
     host = getenv("SNOWFLAKE_TEST_HOST");
+    if (host) 
+    {
+        snowflake_set_attribute(sf, SF_CON_HOST, host);
+    }
+    port = getenv("SNOWFLAKE_TEST_PORT");
+    if (port) 
+    {
+        snowflake_set_attribute(sf, SF_CON_PORT, port);
+    }
+    protocol = getenv("SNOWFLAKE_TEST_PROTOCOL");
+    if (protocol) 
+    {
+        snowflake_set_attribute(sf, SF_CON_PROTOCOL, protocol);
+    }
```

Comment:
Please add setting role based on env, it's common that we would like to select non-default role in OAuth tests:
```c
role = getenv("SNOWFLAKE_TEST_ROLE");
if (role)
{
   snowflake_set_attribute(sf, SF_CON_ROLE, role);
}
```